### ISSUE TITLE
Artefacts

### ISSUE THREAD

**Issue by gexahedron** [geyuying/PF-AFN#5]
Hi!
First - congrats with acceptance to CVPR, great work! and thanks for the code.
Second: I was testing your pretrained model on various pairs of person/cloth, on official demo examples everything is great, however I found various artefacts for other examples (and I didn't find any failure cases in your paper).
Here's what I've noticed so far and have some questions for you, maybe you could help me:
- does the network (and how?) understand what part of the cloth is in front and which should be hidden (behind the neck)? In the image you can see that the back of t-shirt is put on top
![fail1](https://user-images.githubusercontent.com/3427698/110868138-4fdebe80-82c8-11eb-99fa-2c14b9892cb0.jpg)
- pretrained model has problems with arms in front of cloth, sometimes they disappear
![fail6](https://user-images.githubusercontent.com/3427698/110868276-96341d80-82c8-11eb-9c81-e1959aaf9c51.jpg)
sometimes they have some kind of a line inside them
![fail5](https://user-images.githubusercontent.com/3427698/110868326-acda7480-82c8-11eb-91ad-f9af1fe4c0fd.jpg)
- problems with arms, they are morphing into long sleeves (here it's in the left)
![fail4](https://user-images.githubusercontent.com/3427698/110868381-c976ac80-82c8-11eb-8c24-0e926149eb8e.jpg)
- parts of original cloth are still visible
![fail3](https://user-images.githubusercontent.com/3427698/110868452-e4e1b780-82c8-11eb-8681-4e75a53e3d8c.jpg)
or they are preserved in the new cloth, which looks weird
![fail2](https://user-images.githubusercontent.com/3427698/110868488-f4f99700-82c8-11eb-8b23-b37c48675adf.jpg)

Have you studied these or any other cases? Maybe I'm doing something wrong?

**Comment by Gzzgz** [comment]
These are my test results just like your:
![0](https://user-images.githubusercontent.com/15968092/110879591-39f8ea00-8318-11eb-8115-09c96910335e.jpg)
![1](https://user-images.githubusercontent.com/15968092/110879592-3b2a1700-8318-11eb-8b7f-353889c7d146.jpg)
![2](https://user-images.githubusercontent.com/15968092/110879594-3bc2ad80-8318-11eb-8462-684d0a9a5f20.jpg)
![3](https://user-images.githubusercontent.com/15968092/110879598-3c5b4400-8318-11eb-89af-be7006305b4f.jpg)
![4](https://user-images.githubusercontent.com/15968092/110879601-3cf3da80-8318-11eb-9f45-f2335fe3f204.jpg)
![5](https://user-images.githubusercontent.com/15968092/110879603-3d8c7100-8318-11eb-90b1-8d08d825a1b3.jpg)

**Comment by geyuying** [comment]
@Gzzgz Please refer to [https://github.com/geyuying/PF-AFN/issues/3](url) to solve this problem.

**Comment by gexahedron** [comment]
Yes, I wasn't explicit enough, thanks for resolving the issue 3, and the results that I got in the first message here - I get them after fixing the problems, mentioned in the issue 3.
So, I was kind of wondering, could there be maybe any other issues? Or maybe I should preprocess the data at first? Or maybe you're already encountered the problems mentioned above and they are expected from your parser-free model?

**Comment by geyuying** [comment]
@gexahedron 

- does the network (and how?) understand what part of the cloth is in front and which should be hidden (behind the neck)?
Since the training set contain such cases, the network can understand what part of the cloth is in front and which should be hidden such as the case below:

![image](https://user-images.githubusercontent.com/25278412/110921360-70a32480-8359-11eb-9f07-f24c33a070e5.png) ![image](https://user-images.githubusercontent.com/25278412/110921420-7dc01380-8359-11eb-8d44-2546ecf29139.png) ![image](https://user-images.githubusercontent.com/25278412/110921434-81539a80-8359-11eb-8b2e-2111ab800024.png)
  In your listed case, it seems difficult to distinguish between the front and the back, which leads to the failure case, which can sometimes happen during test. In the future work, adding a pixel-level semantic label on the clothes (front or back) may help solve this problem.
-  pretrained model has problems with arms in front of cloth
It is a challenging task in 2D virtual try-on to handle occlusion (for example arms in front of cloth). Our PF-AFN can handle some cases as below:
![image](https://user-images.githubusercontent.com/25278412/110934090-8c61f700-8368-11eb-9913-68d0b2435390.png)  ![image](https://user-images.githubusercontent.com/25278412/110934120-92f06e80-8368-11eb-8349-d8fa57113b4e.png)  ![image](https://user-images.githubusercontent.com/25278412/110934135-98e64f80-8368-11eb-95df-e6c8a9008f71.png)
In the future work, we will exploit the 3D structure of the reference person to handle occlusion in virtual try-on.

As for other cases you listed, they are individual failure cases, which are not common in our test results.
